src/main/java/com/google/jenkins/plugins/computeengine/InstanceConfiguration.java

```diff
@@ -614,22 +618,39 @@ public Instance instance() throws IOException {
     instance.setName(uniqueName());
     instance.setDescription(description);
     instance.setZone(nameFromSelfLink(zone));
+    instance.setMetadata(newMetadata());
+
+    if (!windows) {
+      sshKeyPair = congifureSSHKeyPair(instance, runAsUser);
+    }
 
     if (StringUtils.isNotEmpty(template)) {
       InstanceTemplate instanceTemplate =
           cloud
               .getClient()
               .getTemplate(nameFromSelfLink(cloud.projectId), nameFromSelfLink(template));
+      /* Since we have to set the metadata to include the SSH keypar, we need to ensure
+      we include metadata properties which might be set in the template. */
+      if (instanceTemplate.getProperties() != null
+          && instanceTemplate.getProperties().getMetadata() != null
+          && instanceTemplate.getProperties().getMetadata().getItems() != null) {
+        instanceTemplate
+            .getProperties()
+            .getMetadata()
+            .getItems()
+            .forEach(instance.getMetadata().getItems()::add);
+      }
```

Comment:
On `else` we should not set `clean` metadata?
Im not sure.. but I think we should..

Reply:
It does on line 621.